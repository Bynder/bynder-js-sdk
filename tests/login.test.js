const Bynder = require("../src/bynder-js-sdk.js");
const url = require("url");

const configs = {
    "baseURL": "https://portal.getbynder.com/api/",
    "clientId": "test-client-id",
    "clientSecret": "test-client-secret",
    "redirectUri": "https://test-redirect-uri.com"
};

let bynder;

describe("Make authorization URL", () => {
  bynder = new Bynder(configs);
  const authorizationUrl = bynder.makeAuthorizationURL(
    "state example",
    "offline"
  );

  test("should return correct authorization URL", () => {
    // Security token is random token so we compare without this
    expect(authorizationUrl).toMatchSnapshot();
  });
});

describe("Make API call without token", () => {
  beforeEach(done => {
    const configWithoutToken = {
      baseURL: configs.baseURL,
      clientId: configs.clientId,
      clientSecret: configs.clientSecret,
      redirectUri: configs.redirectUri
    };
    bynder = new Bynder(configWithoutToken);
    bynder
      .getMediaList()
      .then(data => {
        result = data;
        done();
      })
      .catch(error => {
        result = error;
        done();
      });
  });

  test("should return an Error", () => {
    expect(result.message).toEqual("No token found");
  });
});

describe("Make API call with invalid token", () => {
  test("should return an Error", () => {
    const invalidTokenConfig = {
      baseURL: configs.baseURL,
      clientId: configs.clientId,
      clientSecret: configs.clientSecret,
      redirectUri: configs.redirectUri,
      token: { access_token: 2345 }
    };

    expect(() => new Bynder(invalidTokenConfig)).toThrow(Error);
  });
});

describe("Initialize Bynder with permanent token", () => {
  test("should pass", () => {
    const bynder = new Bynder({...configs, permanentToken: "test"});

    expect(bynder.api.permanentToken).toEqual("test");
  });
});
