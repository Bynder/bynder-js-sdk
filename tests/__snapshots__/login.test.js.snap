// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make authorization URL should return correct authorization URL 1`] = `"https://portal.getbynder.com/v6/authentication/oauth2/auth?response_type=code&client_id=test-client-id&redirect_uri=https%3A%2F%2Ftest-redirect-uri.com&scope=offline&state=state%20example"`;
